### PR TITLE
Add github actions

### LINKED ISSUE #279: Migrate to Github Actions
Travis CI is shutting down open source project free credits.

### PR DESCRIPTION
**Pull request by jyapayne**
Fixes #279

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        
+      - name: Cache nimble
+        uses: actions/cache@v2
+        with:
+          path: ~/.nimble
+          key: ${{ runner.os }}-nimble-${{ hashFiles('*.nimble') }}
+
+      - uses: jiro4989/setup-nim-action@v1
+        with:
+          nim-version: "${{ matrix.nim-version }}"
```
Comment (dom96) [inline .github/workflows/main.yml:55]: Do we need this? If we're using choosenim this shouldn't be necessary, right?

**Review by dom96 — APPROVED** [review]
Looks good to me in general. Thank you so much for writing this! Sucks that Travis has been taken over :/

File: .github/workflows/main.yml (hunk 1 above)
Comment (jyapayne) [reply .github/workflows/main.yml:55]: Probably not. I'm new to the whole github actions thing, so could probably remove either one

Comment (dom96) [reply .github/workflows/main.yml:55]: yeah, can you remove the setup-nim-action? :)

**Comment by jyapayne** [comment]
@zetashift congrats on your first PR! 🎉 

The issue for the random test failures is here: https://github.com/nim-lang/nimforum/issues/283

**Pull request merged**